src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -39,6 +40,31 @@ namespace Nethermind.Evm;
 public class VirtualMachine : IVirtualMachine
 {
     public const int MaxCallDepth = 1024;
+    internal static FrozenDictionary<Address, CodeInfo> PrecompileCode { get; } = InitializePrecompiledContracts();
+    internal static LruCache<ValueHash256, CodeInfo> CodeCache { get; } = new(MemoryAllowance.CodeCacheSize, MemoryAllowance.CodeCacheSize, "VM bytecodes");
```

Comment:
Not sure about this cache. If we move it here we share the cache between normal block execution and production and between traced ones which should be mostly RPC. Maybe we should keep those caches separated?

Reply:
Doubled cache instead? Should be cross over between the two, better to share?

Reply:
Not exactly as you can load completely different code in `eth_call` and different in actual processing. Better to keep them separated?

Reply:
Would need to be doing 8k `eth_call`s every 12sec to impact what was previously reserved for block processing; also shouldn't impact actual validators as they shouldn't have RPC open to the world, and would benefit from the larger cache?

As cache hits are 4k to 7k per sec at this level (maybe 8k was too low as cache size)

<img width="966" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/ecd5b2fd-d48e-407b-a842-95b1e6e82249">

Reply:
Code cache size last changed when blockgas limit was 12.5MGas; since that is now upto 30MGas might be time to increase

Reply:
Also it did used to be unified until I made it generic 😅

Reply:
More than halves the reads (+JumpDest calcs)

Top without Mean 80 io/s; Bottom with increased combined cache 30 io/s
<img width="965" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/913f8f1f-2120-4821-aaff-21ec69be95f2">